### PR TITLE
Clarify project naming

### PR DESCRIPTION
**Pull request by reyang**
Follow up based on the votes from https://github.com/open-telemetry/opentelemetry.io/issues/1777#issuecomment-1261330303.

This has been brought up to the Technical Committee meeting, Specification SIG meeting and Maintainers meeting.

This would help to improve consistency in the communication and documentation.

### CODE REVIEW

File: specification/README.md [hunk 1]
```diff
@@ -53,11 +53,16 @@ implementation of the [specification][] is compliant if it satisfies all the
...
 
-## Acronym
+## Project Naming
 
-- The official acronym used by the OpenTelemetry project is "OTel".
-- Refrain from using "OT" in order to avoid confusion with the now deprecated
-  "OpenTracing" project.
+- The official project name is "OpenTelemetry" (with no space between "Open" and
+  "Telemetry").
+- The official acronym used by the OpenTelemetry project is "OTel". Refrain from
```
Comment (yurishkuro) [inline specification/README.md:60]: How about OTEL? I see it used way more often than OTel

```suggestion
- The official acronyms used by the OpenTelemetry project are "OTel" / "OTEL". Refrain from
```

Comment (reyang) [reply specification/README.md:60]: Sounds good to me 👍

I've seen "OTEL" a lot in the environment variables, haven't seen it a lot in docs though.

Comment (yurishkuro) [reply specification/README.md:60]: Yeah, I didn't mean docs but the everyday comms (chats, tickets, etc.) 
It's easier to type all caps than to switch case mid-word.

Comment (arminru) [reply specification/README.md:60]: I usually see `OTel` written in camel case.
Wouldn't `OTEL` suggest an acronym composed of four words as in `HTTP`, `AMQP`, `FWIW`, `LGTM` and the like?
In that sense `OT` would be the right acronym for OpenTelemetry but that one is already taken by OpenTracing.

File: specification/README.md [hunk 2]
```diff
@@ -53,11 +53,16 @@ implementation of the [specification][] is compliant if it satisfies all the
...
+  using "OT" in order to avoid confusion with the now deprecated "OpenTracing"
+  project.
+- The official names for individual implementations follow the pattern of
+  "OpenTelemetry {the name of the programming language or runtime}", for
+  example, "OpenTelemetry Python" and "OpenTelemetry .NET".
```
Comment (svrnm) [inline specification/README.md:65]: what about other components like collector, operator, helm chart, etc? Could this be generalized?

```suggestion
- The official names for sub-projects, like language specific implementations, follow the pattern of
  "OpenTelemetry {the name of the programming language, runtime or component}", for
  example, "OpenTelemetry Python", "OpenTelemetry .NET" or "OpenTelemetry Collector".
```

**Review by arminru — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

File: specification/README.md (hunk 1 above)
Comment (reyang) [reply specification/README.md:60]: For now I'll leave `OTEL` out of the scope of this PR.

**Pull request merged**